app/controllers/ApplicationController.php

```diff
@@ -14,7 +14,7 @@ function __construct(ApplicationForm $applicationForm, SettingRepository $settin
   {
     $this->applicationForm = $applicationForm;
 
-    $this->beforeFilter('currentUser', ['only' => ['edit', 'update']]);
+    $this->beforeFilter('isOwnerOrAdmin', ['only' => ['edit', 'update', 'show']]);
```

Comment:
should be able to use isRole(admin) instead of isOwnerorAdmin. Admin can be it's own filter and owner should also be it's own